File: C++/Ejemplo1/CircunferenciaCentrada.cpp
#include <cmath>
#include <iostream>
#include "CircunferenciaCentrada.h"
using namespace std;
//Constructores
CircunferenciaCentrada::CircunferenciaCentrada()
{
    this->radio = 0.0;
    this->centro = Punto(); //No es necesario, ya que, internamente,
    //en la propia cabecera del método,
    //se ha construido “this->centro”
};
CircunferenciaCentrada::CircunferenciaCentrada(double
                                                   radio,
                                               double
                                                   coord_x,
                                               double coord_y)
{
    this->radio = radio;
    this->centro = Punto(coord_x, coord_y);
    //Internamente ya se ha construido “this->centro” antes de entrar en el
    //cuerpo del constructor.
    //Bastaría con haber hecho “this->centro.setX(coord_x);” y
    //”this->centro.setY(coord_y);”
};

//Métodos
double CircunferenciaCentrada::getRadio()
{
    return this->radio;
};
void CircunferenciaCentrada::setRadio(double nuevo_radio)
{
    this->radio = nuevo_radio;
};
double CircunferenciaCentrada::getDiametro()
{
    return (2 * this->radio);
};
void CircunferenciaCentrada::setDiametro(double nuevo_diametro)
{
    this->radio = nuevo_diametro / 2;
};
double CircunferenciaCentrada::getLongitud()
{
    return (this->radio * 2 * PI);
};
void CircunferenciaCentrada::setLongitud(double nueva_longitud)
{
    this->radio = nueva_longitud / (2 * PI);
};
double CircunferenciaCentrada::getArea()
{
    return (this->radio * this->radio * PI);
};
void CircunferenciaCentrada::setArea(double nuevo_area)
{
    this->radio = sqrt(nuevo_area) / PI;
};
double CircunferenciaCentrada::getXCentro()
{
    return (this->centro.getX());
};
void CircunferenciaCentrada::setXCentro(double nueva_coord_x)
{
    this->centro.setX(nueva_coord_x);
};
double CircunferenciaCentrada::getYCentro()
{
    return this->centro.getY();
};
void CircunferenciaCentrada::setYCentro(double nueva_coord_y)
{
    this->centro.setY(nueva_coord_y);
};

void CircunferenciaCentrada::trasladarCircunferenciaCentrada(double trasl_x,
                                                             double trasl_y)
{
    this->centro.setX(this->centro.getX() + trasl_x);
    this->centro.setY(this->centro.getY() + trasl_y);
};

File: JAVA/circunferenciaCentrada/src/main/java/CircunferenciaCentrada.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author ricky
 */
import java.lang.Math;

public class CircunferenciaCentrada {
    private final static double PI = 3.1416;
    private double radio;
    private Punto centro;
    
    public CircunferenciaCentrada() {
        this.radio = 0.0;
        this.centro = new Punto();
    }
    
    public CircunferenciaCentrada(double radio, double coord_x, double coord_y) {
        this.radio = radio;
        this.centro = new Punto(coord_x, coord_y);
    }
    
    public double getRadio() {
        return this.radio;
    }
    
    public void setRadio(double nueva_radio) {
        this.radio = nueva_radio;
    }
    
    public double getDiametro (){
        return (2 * this.radio);
    }
    public void setDiametro (double nuevo_diametro){
        this.radio = nuevo_diametro / 2;
    }
    public double getLongitud (){
        return (this.radio * 2 * PI);
    }
    public void setLongitud (double nueva_longitud){
        this.radio = nueva_longitud / (2 * PI);
    }
    public double getArea (){
        return (this.radio * this.radio * PI);
    }
    public void setArea (double nuevo_area){
        this.radio = Math.sqrt (nuevo_area )/ PI;
    }
    public double getXCentro (){
        return (this.centro.getX());
    }
    public void setXCentro (double nueva_coord_x){
        this.centro.setX (nueva_coord_x);
    }
    public double getYCentro (){
        return this.centro.getY();
    }
    public void setYCentro (double nueva_coord_y){
        this.centro.setY (nueva_coord_y);
    }
    public void trasladarCircunferenciaCentrada (double trasl_x, double trasl_y){
        this.centro.setX(this.centro.getX() + trasl_x);
    this.centro.setY(this.centro.getY() + trasl_y);
    }
}
    
